File: expressions/Financial_calc.c
// Josy Ramirez, Financial Calculator C

#include <stdio.h>

// Function to get user inputs
void getUserInputs(float *income, float *rent, float *utilities, float *groceries, float *transportation) {
    printf("This is a budget calculator.\n");
    printf("How much do you make a month? ");
    scanf("%f", income);
    printf("How much is your rent? ");
    scanf("%f", rent);
    printf("How much do your utilities cost? ");
    scanf("%f", utilities);
    printf("How much do your groceries cost? ");
    scanf("%f", groceries);
    printf("How much does your transportation cost? ");
    scanf("%f", transportation);
}

// Function to calculate and display percentage breakdown
void calculateAndDisplayPercentages(float income, float rent, float utilities, float groceries, float transportation, float savings, float expenses) {
    printf("\n--- Expense Breakdown (as %% of income) ---\n");
    printf("Rent: %.2f%%\n", (rent / income) * 100);
    printf("Utilities: %.2f%%\n", (utilities / income) * 100);
    printf("Groceries: %.2f%%\n", (groceries / income) * 100);
    printf("Transportation: %.2f%%\n", (transportation / income) * 100);
    printf("Savings: %.2f%%\n", (savings / income) * 100);
    printf("Total Expenses: %.2f%%\n", (expenses / income) * 100);
}

int main() {
    float income, rent, utilities, groceries, transportation;
    
    getUserInputs(&income, &rent, &utilities, &groceries, &transportation);
    
    float savings = income * 0.2;  // 20% of income saved
    float expenses = rent + utilities + groceries + transportation;
    float spend = income - expenses - savings;
    
    // Display summary
    printf("\n--- Budget Summary ---\n");
    printf("Total Income: $%.2f\n", income);
    printf("Total Expenses: $%.2f\n", expenses);
    printf("Savings: $%.2f\n", savings);
    printf("Remaining Money After Expenses and Savings: $%.2f\n", spend);
    
    // Calculate and display percentage breakdown
    calculateAndDisplayPercentages(income, rent, utilities, groceries, transportation, savings, expenses);
    
    printf("Thank you for using my program! :)\n");
    
    return 0;
}

File: expressions/financial_calc.py
# Josy Ramirez, Financial calculator python

# Budget Calculator

def get_user_inputs():
    print("This is a budget calculator.")
    income = float(input("How much do you make a month? "))
    rent = float(input("How much is your rent? "))
    utilities = float(input("How much do your utilities cost? "))
    groceries = float(input("How much do your groceries cost? "))
    transportation = float(input("How much does your transportation cost? "))
    
    return income, rent, utilities, groceries, transportation

def calculate_percentages(income, rent, utilities, groceries, transportation, savings, expenses):
    return {
        "Rent": (rent / income) * 100,
        "Utilities": (utilities / income) * 100,
        "Groceries": (groceries / income) * 100,
        "Transportation": (transportation / income) * 100,
        "Savings": (savings / income) * 100,
        "Total Expenses": (expenses / income) * 100
    }

def main():
    income, rent, utilities, groceries, transportation = get_user_inputs()
    
    savings = income * 0.2  # 20% of income saved
    expenses = rent + utilities + groceries + transportation
    spend = income - expenses - savings
    
    # Display summary
    print("\n--- Budget Summary ---")
    print(f"Total Income: ${income:.2f}")
    print(f"Total Expenses: ${expenses:.2f}")
    print(f"Savings: ${savings:.2f}")
    print(f"Remaining Money After Expenses and Savings: ${spend:.2f}")
    
    # Calculate and display percentage breakdown
    percentages = calculate_percentages(income, rent, utilities, groceries, transportation, savings, expenses)
    print("\n--- Expense Breakdown (as % of income) ---")
    for category, percentage in percentages.items():
        print(f"{category}: {percentage:.2f}%")
    
    print("Thank you for using my program! :)")

if __name__ == "__main__":
    main()
